src/main/java/com/boomi/flow/services/boomi/mdh/common/Entities.java

```diff
@@ -114,50 +121,47 @@ public static MObject createMatchMObject(String universeId, Universe universe, M
         return object;
     }
 
-    private static MObject createAlreadyLinked(Universe universe, Map<String, Object> entity, String idField){
-        List<Property> properties = createPropertiesModel(entity);
+    private static MObject createAlreadyLinked(Universe universe, Multimap<String, Object> entity, String idField){
+        List<Property> properties = createPropertiesModel((Multimap<String, Object>) entity.get(universe.getName()).iterator().next(), null);
```

Comment:
There's a lot of casting and `.iterator().next()` in this PR... is there no way to get rid of them and make it a bit cleaner? They both seem a bit of a smell

Reply:
@jonjomckay  I agree, I just can think in combinations like 
````Map<String, Multimap>```` or ````List<Multimap>```` but I am not sure if that would make it more clean.
The kind of scenarios that need to support are:
````
<element1>	
	<element11>
		<element 112> 
		</element 112>
	</element11>
	<element11>
		<element 112> 
		</element 112>
	</element11>
</element1>
````
and also
````
<element1>	
	<element11>
		<element 112> 
		</element 112>
		<element 112> 
		</element 112>
	</element11>
</element1>
````

Reply:
@jonjomckay I am updating this discussion with some real responses from hub API because the looks a bit different than my simplified examples above (the list are wrapped by other element and by default the name is the same but can be changed):
```
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<RecordQueryResponse resultCount="1" totalCount="1">
    <Record recordId="6433bf10-4a33-4776-adc3-4f9eb172b8e6" createdDate="2019-11-18T14:28:04Z"
            updatedDate="2019-11-18T14:28:04Z">
        <Fields>
            <accountnest2>
                <billing_address>
                    <billing_address>
                        <billing_address_street>street 1</billing_address_street>
                    </billing_address>
                    <billing_address>
                        <billing_address_street>street 2</billing_address_street>
                    </billing_address>
                </billing_address>
            </accountnest2>
        </Fields>
        <links>
            <link source="flow" entityId="1" establishedDate="2019-11-18T14:28:04Z"/>
        </links>
    </Record>
</RecordQueryResponse>
```
and 
```
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<RecordQueryResponse resultCount="1" totalCount="1">
    <Record recordId="3197bfac-9b54-41a2-bf93-034e0bd89027" createdDate="2019-11-18T14:37:43Z" updatedDate="2019-11-18T14:37:43Z">
        <Fields>
            <accountnest3>
                <billing_address>
                    <billing_address_street>
                        <billing_address_street>
                            <name>street 1</name>
                        </billing_address_street>
                        <billing_address_street>
                            <name>street 2</name>
                        </billing_address_street>
                    </billing_address_street>
                </billing_address>
            </accountnest3>
        </Fields>
        <links>
            <link source="flow" entityId="1" establishedDate="2019-11-18T14:37:43Z"/>
        </links>
    </Record>
</RecordQueryResponse>
```